### PR TITLE
Major cleanup of the maven-build

### PR DESCRIPTION
**Pull request by chrisdutz**
-	Nothing is built outside the target directory anymore
o	I moved the static parts of the client and server distributions to “src/assembly/resources” and added an assembly that builds the client and server inside the target directory (You need to update the documentation on this)
o	The sever is now available at server/target/iotdb-server-0.9.0-SNAPSHOT
o	The client is now available at client/target/iotdb-client-0.9.0-SNAPSHOT
o	I updated the integration tests to work with these new locations
-	I updated the distribution module (binary-distribution) to simply include and unpack the client and server assemblies, which greatly simplifies the assembly itself.
-	In the integration-tests you use a process builder to get the current working directory … you can access this in the “user.dir” System property.
-	There was a large mixture of artifacts with different versions in the classpath.
o	I set all versions to the same using a big dependencyManagement block in the master pom
o	I removed the version of every external dependency and moved it into the main dependencyManagement block in the root pom (except dependencies only used in the examples, for these I added a dependencyManagement block in the example pom). This way you instantly identify external dependencies.
o	I added an enforcer rule to fail the build if two differing versions of one artifact are being used in the build (This should eliminate this problem from re-occuring)
o	I replaced the usage of “${project.version}” with the real version (There is no need to manually update these versions and this way when looking at the pom you instantly know the version … keep in mind if an artifact is deployed to maven central, the user doesn’t have the code and therefore he always has to scroll up to the top in order to see which version is used)
-	There were some imports related to Jdk 11 no longer providing some APIs, I moved those imports to a maven profile, which is activated based on the JDK version
-	HDFSInputTest created a file in “spark/test/” … if the test failed, the rat plugin would complain the next time it’s run.

I checked the build with all tests on Oracle-Java 8, Open-JDK 11, Oracle-JDK 12.

### CODE REVIEW

**Comment by JulianFeinauer** [comment]
Hey @chrisdutz, I just checked the branch and did a `mvn clean package -Papache-release`. When I inspect the generated source rlease it still seems to contain jars:

`.//server/iotdb/lib/FastInfoset-1.2.14.jar
.//server/iotdb/lib/mockito-core-2.23.0.jar
.//server/iotdb/lib/powermock-core-2.0.2.jar
.//server/iotdb/lib/javax.annotation-api-1.3.2.jar
.//server/iotdb/lib/istack-commons-runtime-3.0.6.jar
.//server/iotdb/lib/junit-4.12.jar
.//server/iotdb/lib/libthrift-0.9.3.jar
.//server/iotdb/lib/powermock-api-support-2.0.2.jar
.//server/iotdb/lib/powermock-module-junit4-common-2.0.2.jar
.//server/iotdb/lib/antlr-runtime-3.5.2.jar
.//server/iotdb/lib/javax.activation-api-1.2.0.jar
.//server/iotdb/lib/javassist-3.24.0-GA.jar
.//server/iotdb/lib/iotdb-jdbc-0.9.0-SNAPSHOT.jar
.//server/iotdb/lib/powermock-module-junit4-2.0.2.jar
.//server/iotdb/lib/iotdb-jdbc-0.8.0.jar
.//server/iotdb/lib/logback-classic-1.1.11.jar
.//server/iotdb/lib/mockito-all-1.10.19.jar
.//server/iotdb/lib/fastjson-1.2.31.jar
.//server/iotdb/lib/byte-buddy-1.9.3.jar
.//server/iotdb/lib/commons-collections4-4.0.jar
.//server/iotdb/lib/jaxb-runtime-2.4.0-b180725.0644.jar
.//server/iotdb/lib/objenesis-3.0.1.jar
.//server/iotdb/lib/powermock-api-mockito2-2.0.2.jar
.//server/iotdb/lib/guava-21.0.jar
.//server/iotdb/lib/airline-0.8.jar
.//server/iotdb/lib/tsfile-0.8.0.jar
.//server/iotdb/lib/commons-io-2.5.jar
.//server/iotdb/lib/stax-ex-1.8.jar
.//server/iotdb/lib/jaxb-api-2.4.0-b180725.0427.jar
.//server/iotdb/lib/iotdb-server-0.8.0.jar
.//server/iotdb/lib/hamcrest-core-1.3.jar
.//server/iotdb/lib/tsfile-0.9.0-SNAPSHOT.jar
.//server/iotdb/lib/byte-buddy-agent-1.9.3.jar
.//server/iotdb/lib/powermock-reflect-2.0.2.jar
.//server/iotdb/lib/txw2-2.4.0-b180725.0644.jar
.//server/iotdb/lib/snappy-java-1.0.5-M1.jar
.//server/iotdb/lib/slf4j-api-1.7.12.jar
.//server/iotdb/lib/jsr305-3.0.2.jar
.//server/iotdb/lib/service-rpc-0.9.0-SNAPSHOT.jar
.//server/iotdb/lib/javax.inject-1.jar
.//server/iotdb/lib/iotdb-server-0.9.0-SNAPSHOT.jar
.//server/iotdb/lib/commons-lang3-3.8.1.jar
.//server/iotdb/lib/logback-core-1.1.11.jar
.//server/iotdb/lib/service-rpc-0.8.0.jar
.//client/cli/lib/FastInfoset-1.2.14.jar
.//client/cli/lib/slf4j-api-1.7.22.jar
.//client/cli/lib/commons-cli-1.3.1.jar
.//client/cli/lib/javax.annotation-api-1.3.2.jar
.//client/cli/lib/istack-commons-runtime-3.0.6.jar
.//client/cli/lib/junit-4.12.jar
.//client/cli/lib/libthrift-0.9.3.jar
.//client/cli/lib/javax.activation-api-1.2.0.jar
.//client/cli/lib/iotdb-jdbc-0.9.0-SNAPSHOT.jar
.//client/cli/lib/progressbar-0.7.3.jar
.//client/cli/lib/iotdb-client-0.8.0.jar
.//client/cli/lib/iotdb-jdbc-0.8.0.jar
.//client/cli/lib/logback-classic-1.1.11.jar
.//client/cli/lib/mockito-all-1.10.19.jar
.//client/cli/lib/jline-2.14.5.jar
.//client/cli/lib/fastjson-1.2.31.jar
.//client/cli/lib/jaxb-runtime-2.4.0-b180725.0644.jar
.//client/cli/lib/jline-3.10.0.jar
.//client/cli/lib/tsfile-0.8.0.jar
.//client/cli/lib/commons-io-2.5.jar
.//client/cli/lib/jansi-1.17.1.jar
.//client/cli/lib/stax-ex-1.8.jar
.//client/cli/lib/jaxb-api-2.4.0-b180725.0427.jar
.//client/cli/lib/hamcrest-core-1.3.jar
.//client/cli/lib/tsfile-0.9.0-SNAPSHOT.jar
.//client/cli/lib/commons-lang3-3.1.jar
.//client/cli/lib/txw2-2.4.0-b180725.0644.jar
.//client/cli/lib/snappy-java-1.0.5-M1.jar
.//client/cli/lib/iotdb-client-0.9.0-SNAPSHOT.jar
.//client/cli/lib/service-rpc-0.9.0-SNAPSHOT.jar
.//client/cli/lib/logback-core-1.1.11.jar
.//client/cli/lib/service-rpc-0.8.0.jar`

**Comment by chrisdutz** [comment]
You will have to manually delete the client/cli/lib and server/iotdb/lib directories ... they are not managed by the project.

**Comment by JulianFeinauer** [comment]
Indeed @chrisdutz after cleaning the folder it looks fine, thanks!

File: distribution/pom.xml [hunk 1]
```diff
@@ -83,4 +83,18 @@
             </plugin>
         </plugins>
     </build>
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.iotdb</groupId>
+            <artifactId>iotdb-server</artifactId>
+            <version>0.9.0-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (chrisdutz) [reply distribution/pom.xml]: See my comment. Intentionally changed it to the real version everywhere.

[the remark this answers is not in the document]
Comment (Genius-pig) [reply distribution/pom.xml]: I agree with @chrisdutz. Check out other top-level projects like [spark](https://github.com/apache/spark/blob/master/pom.xml) and [flink](https://github.com/apache/flink/blob/master/pom.xml), using explicit version. A user never checks the version by seeing the top of pom.xml.

[the remark this answers is not in the document]
Comment (Genius-pig) [reply distribution/pom.xml]: But using a larger version number is better. For example, `1.0.0-SNAPSHOT`.

File: distribution/pom.xml [hunk 2]
```diff
@@ -83,4 +83,18 @@
...
+            <type>zip</type>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.iotdb</groupId>
+            <artifactId>iotdb-client</artifactId>
+            <version>0.9.0-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (JulianFeinauer) [reply distribution/pom.xml]: @jixuan1989 Chris commented in his mail about that and it seems like there is a best practice to do it like that, so I suggest to keep it like that.

**Pull request merged**